reader_concurrency_semaphore.hh

```diff
@@ -195,18 +201,24 @@ public:
 
     reader_concurrency_semaphore(int count,
             ssize_t memory,
+            sstring name,
             size_t max_queue_length = std::numeric_limits<size_t>::max(),
-            std::function<std::exception_ptr()> raise_queue_overloaded_exception = default_make_queue_overloaded_exception)
+            std::function<void()> prethrow_action = nullptr)
         : _resources(count, memory)
+        , _wait_list(expiry_handler(name))
+        , _name(std::move(name))
         , _max_queue_length(max_queue_length)
-        , _make_queue_overloaded_exception(raise_queue_overloaded_exception) {
-    }
+        , _prethrow_action(prethrow_action)
+    {}
 
     /// Create a semaphore with practically unlimited count and memory.
     ///
     /// And conversely, no queue limit either.
     explicit reader_concurrency_semaphore(no_limits)
-        : reader_concurrency_semaphore(std::numeric_limits<int>::max(), std::numeric_limits<ssize_t>::max()) {
+        : reader_concurrency_semaphore(
+                std::numeric_limits<int>::max(),
+                std::numeric_limits<ssize_t>::max(),
+                format("unlimited RCS @{}", fmt::v5::ptr(this))) {
```

Comment:
I think just `unlimited reader_concurrency_semaphore` will do. We don't expect to see an exception with this ever, since it is only used in tests and it has practically unlimited units so it will never time out. A simpler name will also not require adding the `print.hh` include.

Reply:
Removed address